### PR TITLE
test: fix engine hive of `Invalid Transition Payload Sync`

### LINKED ISSUE #2839: engine hive: Invalid Transition Payload Sync
### Describe the feature

we're currently failing the Invalid Transition checks with invalid payload

the order of events is (for `/Invalid Transition Payload Sync, Invalid GasLimit`):

0. node on last POW block 0x2
1. engine_newPayloadV1 (invalid):, number 0x5 -> SYNCING 
2. FCU to 1.) -> SYNCING
3. engine_newPayloadV1: same block as 1.) -> expects INVALID

apparently in 3.) the detected as INVALID

in reth
1. buffers block + starts pipeline -> SYNCING
2. FCU: pipeline is syncing -> SYNCING
3. buffered again + pipeline is running returns SYNCING

perhaps we should reconsider:

https://github.com/paradigmxyz/reth/blob/d6ccab1ddcf85e31f5bad15b61404dcb44ee41fb/crates/consensus/beacon/src/engine/mod.rs#L546-L548

### Additional context

./hive --client reth --sim ethereum/engine --sim.limit "/Invalid Transition Payload Sync, Invalid GasLimit"

### PR DESCRIPTION
**Pull request by int88**
fixes: #2839

### CODE REVIEW

**Comment by int88** [comment]
@mattsse @Rjected PTAL

File: crates/consensus/beacon/src/engine/mod.rs [hunk 1]
```diff
@@ -849,8 +849,15 @@ where
         let block_hash = block.hash();
         let block_num_hash = block.num_hash();
 
+        let mut lowest_buffered_ancestor = self.lowest_buffered_ancestor_or(block.hash);
```
Comment (Rjected) [inline crates/consensus/beacon/src/engine/mod.rs:852]: right, we only want to check a buffered ancestor if there is one

**Review by Rjected — APPROVED** [review]
lgtm but have not run in hive yet

**Comment by Rjected** [comment]
Nice!
```log
INFO[07-11|15:11:02] hiveproxy started                        container=7c3860f72465 addr=172.17.0.2:8081
INFO[07-11|15:11:02] API: suite started                       suite=0 name=engine-api
INFO[07-11|15:11:02] API: test started                        suite=0 test=1 name="Invalid Transition Payload Sync, Invalid GasLimit (reth)"
INFO[07-11|15:11:03] API: client reth started                 suite=0 test=1 container=ea1f9dcb
INFO[07-11|15:11:09] API: container IP requested              network=bridge container=ea1f9dcb ip=172.17.0.4
INFO[07-11|15:11:12] API: test ended                          suite=0 test=1 pass=true
INFO[07-11|15:11:12] API: suite ended                         suite=0
INFO[07-11|15:11:12] simulation ethereum/engine finished      suites=1 tests=1 failed=0
```

**Pull request merged**